src/pyspiffe/svid/jwt_svid.py

```diff
@@ -1,19 +1,156 @@
+    
+    The JWT-SVID signature is verified using the JWT bundle source.
+
+    Args:
+        token:               a token as a string that is parsed and validated
+        jwt_bundle_source:   an implementation of a {@link JwtBundle} that provides the JWT authorities to
+                                verify the signature
+        audience:            a list of strings used to validate the 'aud' claim
+    
+    Returns:
+        an instance of {@link JwtSvid} with a SPIFFE ID parsed from the 'sub', audience from 'aud', and expiry
+        from 'exp' claim.
+    
+    Raises:
+        JwtSvidError:           when the token expired or the expiration claim is missing,
+                                when the algorithm is not supported, when the header 'kid' is missing,
+                                when the signature cannot be verified, or
+                                when the 'aud' claim has an audience that is not in the audience list
+                                provided as parameter
+        ValueError:             when the token is blank or cannot be parsed
+        BundleNotFoundError:    if the bundle for the trust domain of the spiffe id from the 'sub'
+                                cannot be found the jwt_bundle_source
+        AuthorityNotFoundError: if the authority cannot be found in the bundle using the value from
+                                the 'kid' header
+    """
+
+    @classmethod
+    def parse_and_validate(
+        cls, token: str, jwt_bundle: JwtBundle, audience: []
+    ) -> 'JwtSvid':
+
+        token_header = jwt.get_unverified_header(token)
+        signing_key = jwt_bundle.findJwtAuthority(token_header['kid'])
+
+        claims = jwt.decode(
+            token,
+            algorithms=token_header['alg'],
+            key=signing_key,
+            audience=audience,
+            options={
+                'verify_signature': True,
+                'verify_aud': True,
+                'verify_exp': True,
+                'require': cls._required_claims,
+            },
+        )
+
+        spiffe_ID = SpiffeId.parse(claims['sub'])
+        result = JwtSvid(spiffe_ID, claims['aud'], claims['exp'], claims, token)
+
+        return result
+
+    # If and when https://github.com/jpadilla/pyjwt/issues/599 is fixed, this can be simplified/removed.
+    @classmethod
+    def _validate_exp(cls, expiration_date: str) -> None:
+        expiration_date = int(expiration_date)
+        utctime = timegm(datetime.datetime.utcnow().utctimetuple())
+        if expiration_date < utctime:
+            raise JwtSvidError(SIGNATURE_EXPIRED_ERROR)
+
+    ###
+    ### Verifies if any item specified by audience is present in audience_clains.
+    ### Raises:
+    ###     JwtSvidError: in case none of the items specified by audience is present in audience_claims.
+    ###
+    @classmethod
+    def _validate_aud(cls, audience_claims: [], audience: []) -> None:
+        if not audience_claims and not audience:
```

Comment:
replace by `if not audience` to make it more clear. In case the `audience_claim` is empty it will not pass the validation of being a superset of the expected audience.

Reply:
In this case, the exception raised is different and I wanted it to be consistent: ValueError for empty input. But I am ok if this is going to be consistent with other spiffe libraries.. so, in case audience is specified and claim['aud'] does not exist or is empty, do the other libraries raise "Audience does not match kind of error"? I will check.. if that is the case, I can make the change as you suggest.